src/common/MarkerLabel/MarkerLabel.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC, useCallback } from 'react';
+
+export interface MarkerLabelProps {
+  /**
+   * Width for positioning. Set internally by `Marker`.
+   */
+  width: number;
+
+  /**
+   * Height for positioning. Set internally by `Marker`.
+   */
+  height: number;
+
+  /**
+   * Color of the label.
+   */
+  color: string;
+
+  /**
+   * Position of the label with respect to `Marker` height.
+   */
+  position: string;
+
+  /**
+   * Label text.
+   */
+  text: string;
+
+  /**
+   * Set text to align with y-axis. Set internally by `Marker`.
+   */
+  horizontal: boolean;
+
+  /**
+   * D3 scale for X Axis. Set internally by Marker.
+   */
+  xScale: any;
+
+  /**
+   * D3 scale for Y Axis. Set internally by Marker.
+   */
+  yScale: any;
+}
+
+export const MarkerLabel: FC<Partial<MarkerLabelProps>> = ({
+  height,
+  width,
+  color = '#eee',
+  position = 'center',
+  horizontal,
+  text,
+  xScale,
+  yScale
+}) => {
+  const getPosition = useCallback(() => {
+    let y = yScale(height);
+    let x = width - 70;
+    if (position === 'top') y = yScale(height) - 10;
+    if (position === 'below') y = yScale(height) + 10;
```

Comment:
so `top` and `below` aren't equivalent to me - maybe `top` and `bottom` instead or `above` and `below` but if the line itself is vertical, those also doesn't make sense. We also need to deal with left and right if the marker vertical. 

where do you get the `- 10` and `+ 10` from? would this be an `offset` prop we can pass in?

Reply:
Decided on 'above' and 'below' since I think this works for both line directions, above the line and below the line during horizontal and to the right (above) and to the left (below) of the line for vertical. Also added new offset props to give label more versatility.